api/pkg/cli/cmd/check_upgrade/check_uprade.go

```diff
@@ -0,0 +1,211 @@
+// limitations under the License.
+
+package check_upgrade
+
+import (
+	"strings"
+	"text/template"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/hub/api/pkg/cli/app"
+	"github.com/tektoncd/hub/api/pkg/cli/formatter"
+	"github.com/tektoncd/hub/api/pkg/cli/hub"
+	"github.com/tektoncd/hub/api/pkg/cli/installer"
+	"github.com/tektoncd/hub/api/pkg/cli/kube"
+	"github.com/tektoncd/hub/api/pkg/cli/printer"
+)
+
+const upgradeTemplate = `{{decorate "underline bold" "Hub Tasks\n" }}
+{{- $len := len .HubResources }}{{- if ne $len 0 }}
+NAME	CURRENT_VERSION	NEWER_VERSION
+{{- range $el := .HubResources }}
+{{ $el.Name }}	{{ $el.CurrentVersion }}	{{ $el.LatestVersion }}
+{{- end }}
+{{- else }}
+No Tasks for Upgrade
+{{- end }}
+{{ $len := len .NonHubResources }}{{ if ne $len 0 }}
+{{ decorate "underline bold" "Skipped Tasks\n" }}
+NAME
+{{- range $el := .NonHubResources }}
+{{ icon "bullet" }}{{ $el }}
+{{- end }}
+{{- end }}
+`
+
+var (
+	funcMap = template.FuncMap{
+		"icon":     formatter.Icon,
+		"decorate": formatter.DecorateAttr,
+	}
+	tmpl = template.Must(template.New("Check Upgrade").Funcs(funcMap).Parse(upgradeTemplate))
+)
+
+type hubRes struct {
+	Name           string
+	CurrentVersion string
+	LatestVersion  string
+}
+
+type templateData struct {
+	NonHubResources []string
+	HubResources    []hubRes
+}
+
+const (
+	defaultCatalog = "tekton"
+	versionLabel   = "app.kubernetes.io/version"
+	hubLabel       = "hub.tekton.dev/catalog"
+)
+
+type options struct {
+	cli     app.CLI
+	from    string
+	kind    string
+	version string
+	args    []string
+	kc      kube.Config
+	cs      kube.ClientSet
+}
+
+var cmdExamples string = `
+Check for Upgrades of %S installed via Tekton Hub CLI:
+
+	tkn hub check-upgrades %s
+
+The above command will check for upgrades of %Ss installed via Tekton Hub CLI
+and will skip the %Ss which are not installed by Tekton Hub CLI.
+`
+
+func Command(cli app.CLI) *cobra.Command {
+	opts := &options{cli: cli}
+
+	cmd := &cobra.Command{
+		Use:   "check-upgrade",
+		Short: "Check for upgrades of resources if present",
+		Long:  ``,
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		SilenceUsage: true,
+	}
+	cmd.AddCommand(
+		commandForKind("task", opts),
+	)
+
+	cmd.PersistentFlags().StringVar(&opts.from, "from", defaultCatalog, "Name of Catalog to which resource belongs.")
+
+	cmd.PersistentFlags().StringVarP(&opts.kc.Path, "kubeconfig", "k", "", "Kubectl config file (default: $HOME/.kube/config)")
+	cmd.PersistentFlags().StringVarP(&opts.kc.Context, "context", "c", "", "Name of the kubeconfig context to use (default: kubectl config current-context)")
+	cmd.PersistentFlags().StringVarP(&opts.kc.Namespace, "namespace", "n", "", "Namespace to use (default: from $KUBECONFIG)")
+
+	return cmd
+}
+
+// commandForKind creates a cobra.Command that when run sets
+// opts.Kind and opts.Args and invokes opts.run
+func commandForKind(kind string, opts *options) *cobra.Command {
```

Comment:
should we move this to a separate file? task.go ... for other command too we should do that

Reply:
eventually when we add support for pipeline in commands we will have to do ... so why not start now? wdyt